### ISSUE TITLE
 2.6.0. P2 screen display issue./2.6.0.P2屏幕显示问题。

### ISSUE THREAD

**Issue by djcry** [kingToolbox/WindTerm#1710]
I have two screens
The secondary screen resolution is 1920x1080, the zoom rate is 100%, and everything displays normally when the software is opened.
The main screen resolution is 2560x1600, scaled to 150%, and the problem of font enlargement when opening the software is very serious.
It shouldn't be related to multiple screens, I only keep the main screen, which also exists.
And I haven't found a suitable solution, changing the DPI of the windterm.exe file will make the font more blurry and unable to handle it.
This issue does not exist in 2.6.0.P1. Can you provide a solution to this problem?
我有两个屏幕
副屏幕分辨率为1920x1080，缩放率为100％，打开软件一切显示正常。
主屏幕分辨率为2560x1600,缩放为150％，打开软件字体放大的问题非常严重。
跟多屏幕应该没有关系，我只保留主屏幕这个问题也存在。我推测跟windows的缩放功能有关系。
我没有找到合适的解决方案，更改windterm.exe文件DPI会让字体更加模糊而无法处理。
这个问题在2.6.0.P1里没有存在。这个问题能给个解决方案吗？

**Comment by gooker** [comment]
同样 变大了1.5倍，改dpi可以缩小但是模糊没法用

**Comment by kingToolbox** [comment]
The screen display has indeed changed in the `2.6.0 Prerelease 2` version.

`Qt` offers two possible High-DPI scale factor rounding policies: `passthrough` or `rounding`. Before the `2.6.0 Prerelease 1` version, the `passthrough` policy was used, but many users reported that non-integer scaling factors (such as Windows 125%) would cause the display to appear blurry. Therefore, the policy has been changed to `rounding`. However, it is still apparent that there are issues with multiple displays. I will continue to investigate this issue and see if `Qt` has a better way to solve it.

Currently, Qt 5.15 can only set one policy for the entire application, but it would be better if it could be set per screen. Perhaps this is possible in the new Qt 6.0 version. `WindTerm 3.1.0` will begin to switch to Qt version 6.0.

**Comment by gooker** [comment]
> The screen display has indeed changed in the `2.6.0 Prerelease 2` version.

happy to seeeee you!

**Comment by kingToolbox** [comment]
@gooker 
I'm glad to see you too 😄, it's been a really long time. The main reason is that I've been busy in the past six months, and `WindTerm` updates have been slow, leaving many bugs unfixed and causing trouble for everyone. I am working hard to change this situation, and it should gradually improve.

**Comment by gooker** [comment]
@kingToolbox   my monitor is 4k and 150% dpi, i can modify edit font size small  ,if the menu button.. font size can modify ,i think is ok.
![image](https://github.com/kingToolbox/WindTerm/assets/5688076/798e2a2c-71a7-4728-80da-796b79b4b3d8)

**Comment by lichong5258623** [comment]
I also encountered the same problem，Currently, version 2.6.0. P1 has been rolled back for use.

**Comment by 78740300** [comment]
Can you send a link for version 2.6.0P1?&nbsp;
I can't see the author's version 2.6.0P1 on Github anymore




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "kingToolbox/WindTerm"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年6月27日(星期二) 中午11:10
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [kingToolbox/WindTerm]  2.6.0. P2 screen display issue./2.6.0.P2屏幕显示问题。 (Issue #1710)





 
I also encountered the same problem，Currently, version 2.6.0. P1 has been rolled back for use.

**Comment by gooker** [comment]
> Can you send a link for version 2.6.0P1?&nbsp; I can't see the author's version 2.6.0P1 on Github anymore
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "kingToolbox/WindTerm" ***@***.***&gt;; 发送时间:&nbsp;2023年6月27日(星期二) 中午11:10 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [kingToolbox/WindTerm] 2.6.0. P2 screen display issue./2.6.0.P2屏幕显示问题。 (Issue #1710) I also encountered the same problem，Currently, version 2.6.0. P1 has been rolled back for use. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

https://github.com/kingToolbox/WindTerm/releases/tag/2.6-prerelease

**Comment by 78740300** [comment]
There is no download link for 2.6.0.P1 anymore





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "kingToolbox/WindTerm"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年6月28日(星期三) 上午10:27
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [kingToolbox/WindTerm]  2.6.0. P2 screen display issue./2.6.0.P2屏幕显示问题。 (Issue #1710)





  
Can you send a link for version 2.6.0P1?&nbsp; I can't see the author's version 2.6.0P1 on Github anymore
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "kingToolbox/WindTerm" @.&gt;; 发送时间:&nbsp;2023年6月27日(星期二) 中午11:10 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [kingToolbox/WindTerm] 2.6.0. P2 screen display issue./2.6.0.P2屏幕显示问题。 (Issue #1710) I also encountered the same problem，Currently, version 2.6.0. P1 has been rolled back for use. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: @.***&gt;
  
https://github.com/kingToolbox/WindTerm/releases/tag/2.6-prerelease

**Comment by tangqiu0205** [comment]
I also encountered the same problem.

**Comment by hisune** [comment]
I tried the 2.6.0 prerelease_2 version and I also had this problem, and I have rolled back to the 2.5.0 version to continue using it.

**Comment by PerfBleu** [comment]
write a cmd script in the folder that WindTerm installed, write content below in the cmd file:
```
set QT_SCALE_FACTOR_ROUNDING_POLICY=PassThrough
WindTerm.exe
```
Save these command in run.cmd, and start windterm from it.
then, the problem solved.

**Comment by 78740300** [comment]
Running through cmd is effective, but running windterm.exe directly still has issues.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "kingToolbox/WindTerm"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年9月15日(星期五) 下午4:23
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [kingToolbox/WindTerm]  2.6.0. P2 screen display issue./2.6.0.P2屏幕显示问题。 (Issue #1710)





 
write a cmd script in the folder that WindTerm installed, write content below in the cmd file:
 set QT_SCALE_FACTOR_ROUNDING_POLICY=PassThrough WindTerm.exe  
Save these command in run.cmd, and start windterm from it.
 then, the problem solved.

**Comment by PerfBleu** [comment]
> Running through cmd is effective, but running windterm.exe directly still has issues.
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "kingToolbox/WindTerm" ***@***.***&gt;; 发送时间:&nbsp;2023年9月15日(星期五) 下午4:23 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [kingToolbox/WindTerm] 2.6.0. P2 screen display issue./2.6.0.P2屏幕显示问题。 (Issue #1710) write a cmd script in the folder that WindTerm installed, write content below in the cmd file: set QT_SCALE_FACTOR_ROUNDING_POLICY=PassThrough WindTerm.exe Save these command in run.cmd, and start windterm from it. then, the problem solved. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

yes, because you need to set the environment varaible. You can also add it to your system environment varaibles, but it may effect other QT applications
